libraries/rush-lib/src/logic/Git.ts

```diff
@@ -368,26 +425,307 @@ export class Git {
+            getFieldAndAdvancePos(' ');
+          state = GitStatusEntryState.indexFileMode;
+          break;
+        }
+
+        case GitStatusEntryState.indexFileMode: {
+          (currentObject as IChangedGitStatusEntry | IRenamedOrCopiedGitStatusEntry).indexFileMode =
+            getFieldAndAdvancePos(' ');
+          state = GitStatusEntryState.worktreeFileMode;
+          break;
+        }
+
+        case GitStatusEntryState.stage1FileMode: {
+          (currentObject as IUnmergedGitStatusEntry).stage1FileMode = getFieldAndAdvancePos(' ');
+          state = GitStatusEntryState.stage2FileMode;
+          break;
+        }
+
+        case GitStatusEntryState.stage2FileMode: {
+          (currentObject as IUnmergedGitStatusEntry).stage2FileMode = getFieldAndAdvancePos(' ');
+          state = GitStatusEntryState.stage3FileMode;
+          break;
+        }
+
+        case GitStatusEntryState.stage3FileMode: {
+          (currentObject as IUnmergedGitStatusEntry).stage3FileMode = getFieldAndAdvancePos(' ');
+          state = GitStatusEntryState.worktreeFileMode;
+          break;
+        }
+
+        case GitStatusEntryState.worktreeFileMode: {
+          (currentObject as IChangedGitStatusEntry | IRenamedOrCopiedGitStatusEntry).worktreeFileMode =
+            getFieldAndAdvancePos(' ');
+
+          if (isUnmerged) {
+            state = GitStatusEntryState.stage1ObjectName;
+          } else {
+            state = GitStatusEntryState.headObjectName;
+          }
+
+          break;
+        }
+
+        case GitStatusEntryState.headObjectName: {
+          (currentObject as IChangedGitStatusEntry | IRenamedOrCopiedGitStatusEntry).headObjectName =
+            getFieldAndAdvancePos(' ');
+          state = GitStatusEntryState.indexObjectName;
+          break;
+        }
+
+        case GitStatusEntryState.indexObjectName: {
+          (currentObject as IChangedGitStatusEntry | IRenamedOrCopiedGitStatusEntry).indexObjectName =
+            getFieldAndAdvancePos(' ');
+          if (isRenamedOrCopied) {
+            state = GitStatusEntryState.renameOrCopyScore;
+          } else {
+            state = GitStatusEntryState.path;
+          }
+          break;
+        }
+
+        case GitStatusEntryState.stage1ObjectName: {
+          (currentObject as IUnmergedGitStatusEntry).stage1ObjectName = getFieldAndAdvancePos(' ');
+          state = GitStatusEntryState.stage2ObjectName;
+          break;
+        }
+
+        case GitStatusEntryState.stage2ObjectName: {
+          (currentObject as IUnmergedGitStatusEntry).stage2ObjectName = getFieldAndAdvancePos(' ');
+          state = GitStatusEntryState.stage3ObjectName;
+          break;
+        }
+
+        case GitStatusEntryState.stage3ObjectName: {
+          (currentObject as IUnmergedGitStatusEntry).stage3ObjectName = getFieldAndAdvancePos(' ');
+          state = GitStatusEntryState.path;
+          break;
+        }
+
+        case GitStatusEntryState.renameOrCopyScore: {
```

Comment:
With `--no-renames` this field will never show up.

Reply:
I like leaving the renames here, in case this gets used for something more complex in the future.